app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,69 @@
+require 'net/http'
+require 'json'
+
+POSTCODE_BASE_API = "https://api.postcodes.io/postcodes/"
+TFL_BASE_API = "https://api.tfl.gov.uk/StopPoint/"
+
+
+
+class PagesController < ApplicationController
+  def index
+  end
+
+  def postcode
+    lat,lon = get_coords_by_postcode
+    if lat == nil && lon == nil
+      flash[:error] = "Invalid postcode. Please try again."
+      redirect_to root_path
+    else
+      stop_points_information = get_stop_points_by_lat_lon(lat, lon)
+      if stop_points_information == nil
+        flash[:error] = "No bus stations nearby"
+        redirect_to root_path
+      else
+        @stops = []
+        @length = []
+        stop_points_information.each { |stop_point|
+          unless stop_point["lines"].empty?
+            bus_departures = fetch_arrivals(stop_point)
+            stop_name = stop_point["commonName"]
+            stop_letter = stop_point["stopLetter"]
+            name = "#{stop_name}#{stop_letter ? " - Stop #{stop_letter}" : "" }:"
```

Comment:
I like the splitting, maybe we can also put this into a separate method, e.g. `format_stop_name`? Then we don't need to be concerned with what's happening, just that it's being formatted and we can assume it's being done correctly